packages/flutter/test/widgets/text_test.dart

```diff
@@ -136,4 +138,114 @@ void main() {
     expect(semantics, hasSemantics(expectedSemantics, ignoreTransform: true, ignoreId: true, ignoreRect: true));
     semantics.dispose();
   });
+
+  testWidgets('recognizers split semantic node', (WidgetTester tester) async {
+    final SemanticsTester semantics = new SemanticsTester(tester);
+    const TextStyle textStyle = TextStyle(fontFamily: 'Ahem');
+    await tester.pumpWidget(
+      new Text.rich(
+        new TextSpan(
+          children: <TextSpan>[
+            const TextSpan(text: 'hello '),
+            new TextSpan(text: 'world', recognizer: new TapGestureRecognizer()..onTap = () {}),
+            const TextSpan(text: ' this is a '),
+            const TextSpan(text: 'cat-astrophe'),
+          ],
+          style: textStyle,
+        ),
+        textDirection: TextDirection.ltr,
+      ),
+    );
+    final TestSemantics expectedSemantics = new TestSemantics.root(
+      children: <TestSemantics>[
+        new TestSemantics.rootChild(
+          rect: new Rect.fromLTRB(0.0, 0.0, 800.0, 600.0),
+          children: <TestSemantics>[
+            new TestSemantics(
+              rect: new Rect.fromLTRB(-4.0, -4.0, 88.0, 18.0),
+              label: 'hello ',
+              textDirection: TextDirection.ltr,
+            ),
+            new TestSemantics(
+              rect: new Rect.fromLTRB(80.0, -4.0, 158.0, 18.0),
+              label: 'world',
+              textDirection: TextDirection.ltr,
+              actions: <SemanticsAction>[
+                SemanticsAction.tap,
+              ],
+            ),
+            new TestSemantics(
+              rect: new Rect.fromLTRB(150.0, -4.0, 480.0, 18.0),
+              label: ' this is a cat-astrophe',
+              textDirection: TextDirection.ltr,
+            )
+          ],
+        ),
+      ],
+    );
+    expect(semantics, hasSemantics(expectedSemantics, ignoreTransform: true, ignoreId: true));
+    semantics.dispose();
+  });
+
+  testWidgets('recognizers split semantic node - bidi', (WidgetTester tester) async {
+    final SemanticsTester semantics = new SemanticsTester(tester);
+    const TextStyle textStyle = TextStyle(fontFamily: 'Ahem');
+    await tester.pumpWidget(
+      new RichText(
+        text: new TextSpan(
+          style: textStyle,
+          children: <TextSpan>[
+            const TextSpan(text: 'hello world${Unicode.RLE}${Unicode.RLO} '),
+            new TextSpan(text: 'BOY', recognizer: new LongPressGestureRecognizer()..onLongPress = () {}),
+            const TextSpan(text: ' HOW DO${Unicode.PDF} you ${Unicode.RLO} DO '),
+            new TextSpan(text: 'SIR', recognizer: new TapGestureRecognizer()..onTap = () {}),
+            const TextSpan(text: '${Unicode.PDF}${Unicode.PDF} good bye'),
+          ],
+        ),
+        textDirection: TextDirection.ltr,
+      )
+    );
+    final TestSemantics expectedSemantics = new TestSemantics.root(
```

Comment:
Maybe add as a comment here in what order the text will appear on the screen to make the following easier to understand?